nova/core/roles/docker/tasks/debian_os.yml

```diff
@@ -15,7 +15,7 @@
   block:
     - name: Adding Docker GPG apt key...
       ansible.builtin.apt_key:
-        url: https://download.docker.com/linux/debian/gpg
```

Comment:
This needs to be hardcoded because it's for Kali that does not have a separate official docker repo, so Debian is used.

Reply:
I would then suggest to modify this task in `defaults/debian_os.yml`:
```yaml
- name: Adding Docker GPG apt key...
  ansible.builtin.apt_key:
    url: "{{ docker_apt_proxy }}/gpg"
    keyring: /etc/apt/trusted.gpg.d/docker.gpg
    state: present
  register: docker_apt_key
  until: not docker_apt_key.failed # Because sometimes the primary DNS is not up yet or egress FW is still being deployed
  retries: 10
  delay: 6
  vars:
    # Kali does not have a seperate official docker repo
    distribution: "debian"
```

and in defaults/main.yml
```yaml
docker_apt_proxy: https://download.docker.com/linux/{{ distribution }}
distribution: "{{ ansible_distribution | lower }}"
```

Reply:
Yes that would work

Reply:
I have just implemented it